server-spi-private/src/main/java/org/keycloak/userprofile/DefaultUserProfile.java

```diff
@@ -113,6 +121,11 @@ private UserModel updateInternal(UserModel user, boolean removeAttributes, BiCon
 
                 if (currentValue.size() != updatedValue.size() || !currentValue.containsAll(updatedValue)) {
                     user.setAttribute(name, updatedValue);
+                    
+                    if(UserModel.EMAIL.equals(name) && RESET_EMAIL_VERIFIED_CONTEXTS.contains(metadata.getContext())) {
```

Comment:
I think this type of logic is also solved by what we are discussing around listening for attribute changes.

We could probably work on this capability prior to introducing this logic to reset email verified.

Reply:
Or we can merge this PR and then move this logic into the attribute change listener as an example of its use in the listener PR. Whatever works for you, I don't mind too much (but hope listener will be here soon not to have too much conflicts in this PR while it waits for listeners ;-)